### PR TITLE
ci: use multiple stage

### PR DESCRIPTION
**Pull request by lizan**
To be merged along with #13550. Mostly cosmetic change to use multiple stage so it is easier to see / retest in AZP UI. Also made docs as part of precheck so it blocks CI.

Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

**Comment by phlax** [comment]
@lizan one thing i just realized is that some of the docs/READMEs/scripts refer to CircleCI - eg 

https://github.com/envoyproxy/envoy/blob/master/docs/README.md

doing `git grep -i circleci` it seems like there are a few more

**Comment by lizan** [comment]
@phlax https://github.com/envoyproxy/envoy/pull/13558

**Comment by mattklein123** [comment]
OK when this merges I will need to change all the required checks. Can you merge main?

/wait

**Comment by lizan** [comment]
@mattklein123 this is good to go, but lmk if you want do any cosmetic change (since it affects required checks name).

File: .azure-pipelines/pipelines.yml
Comment (mattklein123) [inline .azure-pipelines/pipelines.yml:185]: Is there a way to remove "check" from the job name so they match what we have today? I don't think it adds much and it would be nice to not have to change the required jobs?

/wait-any

**Comment by mattklein123** [comment]
Yeah I think my main comment is can we avoid changing the job names or is that impossible?

If we have to change the job names, can we satisfy my OCD by making them all lower case or all properly capitalized? Right now it's all over the place. Thank you!

/wait

**Comment by lizan** [comment]
> Yeah I think my main comment is can we avoid changing the job names or is that impossible?

That will mess up the stage view if we made empty:
image

> If we have to change the job names, can we satisfy my OCD by making them all lower case or all properly capitalized? Right now it's all over the place. Thank you!

totally agreed and that it will satisfy my OCD either :joy:

**Comment by mattklein123** [comment]
Thanks this LGTM. Do we want to sort out why OSX is still perma failing it seems before merging this?

/wait-any

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
